From e82794ea7290597446d80c87033750c8e45264f0 Mon Sep 17 00:00:00 2001
From: Vinith Kumar R <quic_vinithku@quicinc.com>
Date: Thu, 12 Sep 2024 17:27:47 +0530
Subject: [PATCH] wifi: ath12k: avoid filling rx status for invalid band index

Currently, in ath12k_dp_mon_rx_update_peer_stats_ds function, we are
updating the rx stats even when rx_status->band reaches a maximum
index value 6. Due to this, observing crash in sta_stats_decode_rate
function while accessing sband->bitrates. So avoid updating the
rx stats in the maximum band value case.

Fixes: I5620c5b39 (QSDK: ath12k: Fix dead lock issue with ath12k_dp_mon_rx_update_peer_stats_ds)

Patch-dependency: 972-QSDK-ath12k-Fix-dead-lock-issue-with-ath12k_dp_mon_r.patch

Signed-off-by: Vinith Kumar R <quic_vinithku@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_mon.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/dp_mon.c b/drivers/net/wireless/ath/ath12k/dp_mon.c
index f4e9401..e0d6efd 100644
--- a/drivers/net/wireless/ath/ath12k/dp_mon.c
+++ b/drivers/net/wireless/ath/ath12k/dp_mon.c
@@ -3392,6 +3392,9 @@ static void ath12k_dp_mon_rx_update_peer_stats_ds(struct ath12k *ar,
 			rx_status = &status;
 		}
 
+		if (rx_status->band == NUM_NL80211_BANDS)
+			return;
+
 		sta = container_of((void *)arsta->ahsta, struct ieee80211_sta, drv_priv);
 #ifdef CPTCFG_MAC80211_DS_SUPPORT
 		ieee80211_rx_update_stats(ar->ah->hw, sta, arsta->link_id,
@@ -3445,6 +3448,9 @@ static void ath12k_dp_mon_rx_update_peer_stats_ds(struct ath12k *ar,
 			rx_status = &status;
 		}
 
+		if (rx_status->band == NUM_NL80211_BANDS)
+			continue;
+
 		sta = container_of((void *)arsta->ahsta, struct ieee80211_sta, drv_priv);
 #ifdef CPTCFG_MAC80211_DS_SUPPORT
 		ieee80211_rx_update_stats(ar->ah->hw, sta, arsta->link_id,
-- 
2.34.1

